daemon/algod/server.go

```diff
@@ -227,30 +225,26 @@ func (s *Server) Start() {
 	c := make(chan os.Signal)
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
 	signal.Ignore(syscall.SIGHUP)
-	go func() {
-		sig := <-c
+
+	fmt.Printf("Node running and accepting RPC requests over HTTP on port %v. Press Ctrl-C to exit\n", addr)
+	select {
+	case err := <-errChan:
+		if err != nil {
+			s.log.Warn(err)
+		} else {
+			s.log.Info("Node exited successfully")
+		}
+		s.Stop()
+	case sig := <-c:
 		fmt.Printf("Exiting on %v\n", sig)
 		s.Stop()
 		os.Exit(0)
-	}()
-
-	fmt.Printf("Node running and accepting RPC requests over HTTP on port %v. Press Ctrl-C to exit\n", addr)
-	err = <-errChan
-	if err != nil {
-		s.log.Warn(err)
-	} else {
-		s.log.Info("Node exited successfully")
 	}
 }
 
 // Stop initiates a graceful shutdown of the node by shutting down the network server.
 func (s *Server) Stop() {
-	s.stopping.Lock()
-	defer s.stopping.Unlock()
-
-	if s.stopped {
-		return
-	}
+	close(s.stopping)
```

Comment:
Is this where the handler is notified to shut down?
Please add a comment if so.

Reply:
yes; I will add a comment.